nucliadb/nucliadb/ingest/maindb/pg.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import annotations
+
+import asyncio
+from typing import Any, AsyncGenerator, List, Optional
+
+import asyncpg
+
+from nucliadb.ingest.maindb.driver import DEFAULT_SCAN_LIMIT, TXNID, Driver, Transaction
+from nucliadb.ingest.maindb.exceptions import NoWorkerCommit
+
+CREATE_TABLE = """
+CREATE TABLE IF NOT EXISTS resources (
+    key TEXT PRIMARY KEY,
+    value BYTEA
+);
+"""
+
+
+class DataLayer:
+    def __init__(self, connection: asyncpg.Connection):
+        self.connection = connection
+
+    async def get(self, key: str) -> Optional[bytes]:
+        return await self.connection.fetchval(
+            "SELECT value FROM resources WHERE key = $1", key
+        )
+
+    async def set(self, key: str, value: bytes) -> None:
+        await self.connection.execute(
+            """
+INSERT INTO resources (key, value)
+VALUES ($1, $2)
+ON CONFLICT (key)
+DO UPDATE SET value = EXCLUDED.value
```

Comment:
How are conflicts handling working here?

Is this line telling pg that on conflict needs to update the conflict table?
Will it bubble up any exceptions?